cpp/src/join/conditional_join_kernels.cuh

```diff
@@ -242,9 +245,13 @@ __global__ void conditional_join(table_device_view left_table,
                         join_shared_r[warp_id]);
     }
 
+    __syncwarp(activemask);
```

Comment:
It would be great to know why we need to sync all the treads before and after this cache write

Reply:
Pretty sure It is required to ensure `current_idx_shared[warp_id]` is valid.